doc/cephfs/quota.rst

```diff
@@ -74,6 +74,12 @@ To set a quota::
   setfattr -n ceph.quota.max_bytes -v 100000000 /some/dir     # 100 MB
   setfattr -n ceph.quota.max_files -v 10000 /some/dir         # 10,000 files
 
+``ceph.quota.max_bytes`` can also be set using human-friendly units
+(both SI prefixes and the IEC prefixes i.e. K, M, ... E and Ki, Mi, ... Ei)::
```

Comment:
nit:
```suggestion
(both SI prefixes and the IEC prefixes i.e. K, M, ... E and Ki, Mi ... Ei respectively)::
```

Reply:
Thanks for the review, looking at this doc after long and I think the sentence `both SI prefixes and the IEC prefixes i.e. K, M, ... E and Ki, Mi ... Ei` looks misleading here, this would indicate that using e.g. 10K would mean 10 kilobytes while 10Ki would mean 10 kebibytes but that is wrong, `strict_iec_cast` that is used here to convert human readable form into bytes uses shift left to do the power of 2. This needs to be changed to just IEC prefixes, what do you think @mchangir @rishabh-d-dave

Reply:
@mchangir @rishabh-d-dave @vshankar Adding to the above comment, function `strict_iec_cast()` treats `K` same as `Ki`, it won't throw any warning/error and would just continue with the conversion, so 1K = 1Ki = 1024 bytes. This would be a bit misleading, I think either we modify `strict_iec_cast()` to throw an error on receiving SI units or we just ignore it and note this in the docs, what's your thought?

PS: We do have a function `strict_si_cast()` that checks only for SI units, it has better unit recognition mechanism than `strict_iec_cast()`, we can either use it to identify SI unit or can modify `strict_iec_cast()` to recognize unit just as `strict_si_cast()` does.

Reply:
Don't modify the existing `strict_xxx_cast()` functions
Instead, write a wrapper function (there probably is one already) to test for the `iB` suffix to decide to call `strict_iec_cast()`, if not then call `strict_si_cast()`

Reply:
but question is do we want to allow casting to SI values? Reason I'm asking is usually SI units are for data throughput while IEC units are for file sizes. Do we follow that norm?

Reply:
just search for `strict_si_cast()` and `strict_iec_cast()` in the code and you'll get to know where and why they have been used

but I agree about IEC sizes being more sensible for specifying file sizes

Reply:
I was looking at instances where these functions are used, I think using `strict_iec_cast()` seems rational here, I could find one such relevant instance in [rgw_admin.cc](https://github.com/ceph/ceph/blob/main/src/rgw/rgw_admin.cc#L3643), i.e. for sizes. Rest all instances are some generalized uses of both the functions.

Reply:
an interesting comment in the function `strict_iec_cast` https://github.com/ceph/ceph/blob/main/src/common/strtol.cc#L149-L150, looks like this function was made this way, i.e. to accept both `KiB` and `K` as IEC units.